tests/main/invariant-tool/task.yaml

```diff
@@ -0,0 +1,53 @@
+summary: invariant-tool is detecting problems
+prepare: |
+    truncate --size=0 defer.sh
+    chmod +x defer.sh
+restore: |
+    # Restore system to the previous state by running deferred commands in
+    # reverse order.
+    tac defer.sh > refed.sh
+    sh -xe refed.sh && rm -f {defer,refed}.sh
+execute: |
+    # Invariant tool presents the usage screen when invoked without arguments
+    invariant-tool | MATCH 'usage: invariant-tool check \[INVARIANT...\]'
+
+    # When invoked with "check" argument, it runs either all invariants
+    # or the specified set of invariants.
+    invariant-tool check | MATCH 'invariant-tool: root-files-in-home ok'
+    invariant-tool check root-files-in-home | MATCH 'invariant-tool: root-files-in-home ok'
+
+    # Unknown invariant names are reported.
+    invariant-tool check foo 2>&1 | MATCH 'invariant-tool: unknown invariant foo'
+    not invariant-tool check foo 2>/dev/null
+
+    # The root-files-in-home invariant detects files owned by root anywhere in /home/
+    invariant-tool check root-files-in-home | MATCH 'invariant-tool: root-files-in-home ok'
+    touch /home/test/invariant-canary
+    echo 'rm -f /home/test/invariant-canary' >> defer.sh
+    chown root /home/test/invariant-canary
+    invariant-tool check root-files-in-home 2>&1 | MATCH 'invariant-tool: root-files-in-home not-ok'
+    invariant-tool check root-files-in-home 2>&1 | MATCH 'invariant-tool: the following files should not be owned by root'
+    invariant-tool check root-files-in-home 2>&1 | MATCH '/home/test/invariant-canary'
+    invariant-tool check root-files-in-home 2>&1 | MATCH 'invariant-tool: system is corrupted'
+    not invariant-tool check root-files-in-home 2>/dev/null
+
+    # Chown is the solution.
+    chown test /home/test/invariant-canary
+    invariant-tool check root-files-in-home | MATCH 'invariant-tool: root-files-in-home ok'
+
+    # The root-files-in-home invariant ignores /home/gopath which comes from spread.yaml
+    test -d /home/gopath
+    test "$(stat -c %u /home/gopath)" -eq 0
```

Comment:
thanks for testing this too so when we stop doing this we will remember to update the invariant check too